File: 3375/main.go
package main

import "fmt"

func main() {
	fmt.Println(minOperations([]int{9,7,5,3}, 1))	// 4
	fmt.Println(minOperations([]int{2,1,2}, 2))		// -1
	fmt.Println(minOperations([]int{5,2,5,4,5}, 2))	// 2
}

func minOperations(nums []int, k int) int {

	m := map[int]bool{}
	for _, num := range nums {
		if num < k {
			return -1
		} else if num != k {
			m[num] = true
		}
	}

	return len(m)
}

File: 3375/main.rs
use std::collections::HashSet;

struct Solution {}

impl Solution {
    pub fn min_operations(nums: Vec<i32>, k: i32) -> i32 {
        let mut st = HashSet::new();
        for x in nums {
            if x < k {
                return -1
            } else if x > k {
                st.insert(x);
            }
        }
        st.len() as i32
    }
}

fn main() {
    println!("{}", Solution::min_operations(vec![2,1,2], 2));
}